packages/test/mocha-test-setup/src/mochaHooks.ts

```diff
@@ -24,112 +24,119 @@ const propsDict =
 	send: () => {},
 	flush: async () => {},
 };
 
+// Keep references to the original console functions so we can restore them after each test.
 const log = console.log;
 const error = console.log;
 const warn = console.warn;
-let currentTestLogger: ITelemetryBufferedLogger | undefined;
-let currentTestName: string | undefined;
-let originalLogger: ITelemetryBufferedLogger;
+
+const originalLogger = _global.getTestLogger?.() ?? nullLogger;
+const testLogger = new FluidTestRunLogger(originalLogger);
 
 /**
  * @internal
  */
 export const mochaHooks = {
 	beforeAll() {
-		originalLogger = _global.getTestLogger?.() ?? nullLogger;
-		_global.getTestLogger = () => {
-			// If a current test logger exists, use that. Otherwise, create a new one. This should become the
-			// current test logger if this function is running in a context which understands the current test.
-			// Otherwise, just return the created TestLogger. (This happens e.g. if someone calls `getTestLogger`
-			// in a `before` or `after` hook, due to the order in which mocha hooks run)
-			if (currentTestLogger !== undefined) {
-				return currentTestLogger;
-			}
-
-			const testLogger = new TestLogger(originalLogger, currentTestName);
-			if (currentTestName !== undefined) {
-				currentTestLogger = testLogger;
-			}
-
-			return testLogger;
-		};
+		// Code in our tests will call the global `getTestLogger` function to get a logger to use.
+
+		// First we call the version of that function that was (potentially) injected dynamicaly to get the logger that it
+		// provides and save a reference to it (above).
+		// Then we redefine `getTestLogger` so it returns a more intelligent logger which adds test-run-related context to
+		// all events logged through it. See the documentation on `FluidTestRunLogger` for details.
+		// NOTE: if we have other global mocha hooks defined somewhere, they include a `beforeEach` hook, and the module in
+		// which they are defined gets loaded before this one, then if that `beforeEach` hook calls `getTestLogger` the logger
+		// it will get will still have a lot of the test-run-related context, but not the name of the current test, because
+		// it will run before the `beforeEach` hook in this file which sets that.
+		_global.getTestLogger = () => testLogger;
 	},
 	beforeEach(this: Mocha.Context) {
-		// Suppress console.log if not verbose mode
+		// If not in verbose mode, suppress console output while the current test runs.
 		if (process.env.FLUID_TEST_VERBOSE === undefined) {
 			console.log = () => {};
 			console.error = () => {};
 			console.warn = () => {};
 		}
-		// save the test name can and clear the previous logger (if afterEach didn't get ran and it got left behind)
-		currentTestName = this.currentTest?.fullTitle();
-		currentTestLogger = undefined;
 
-		// send event on test start
-		originalLogger.send({
+		testLogger.setCurrentTest(this.currentTest?.fullTitle() ?? "");
+		testLogger.send({
```

Comment:
This is the main fix. Use the same logger for `Test_start` and `Test_end` that we use for everything else, not just the original logger with no extra test-related context.